### PR TITLE
feat: migrate pages files

### PR DESCRIPTION
**Pull request by dennissiq**
## Proposed Changes
- Add types in all necessary places
- #830 
- #828 
- #829

### CODE REVIEW

**Review by baires — CHANGES_REQUESTED** [review]
Looking good but we need to fix its-friday class to respect the current day

File: pages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        <link
+          rel="icon"
+          type="image/png"
+          href={shouldIDeployFavIcon(now)}
+          sizes="32x32"
+        />
+        <meta property="og:image" content={shouldIDeployAnswerImage(now)} />
+        <title>Should I Deploy Today?</title>
+      </Head>
+      <div className={`wrapper ${!shouldIDeploy(now) && 'its-friday'}`}>
```
[the remark this answers is not in the document]
Comment (dennissiq) [reply pages/index.tsx:54]: @baires can you write a suggestion to solve this? In this file, I've maintained the same logic of `pages.js`

[the remark this answers is not in the document]
Comment (baires) [reply pages/index.tsx:54]: Let me look at this over the weekend

**Comment by baires** [comment]
#### 📝 Changed routes:


| [**/**](https://shouldideploy-ncymgwlk5-sgavel.vercel.app/) |
|:-:|
|<a href="https://shouldideploy-ncymgwlk5-sgavel.vercel.app/">Screenshot of /</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/HqAlgowH9I3-qLbUVE-4VZ_PGGtQMJOYABr5iOaf35DMP1d4lsLqc15lW7_SCaeTJIs9Ii4NG57VLZsujfH83g.png">(view full size)</a> |





1 deleted route:
- **/**

Commit e49fef6389cb02b052f2227bc51d8f50d37023c9 (https://shouldideploy-ncymgwlk5-sgavel.vercel.app).

File: pages/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (baires) [reply pages/index.tsx:54]: Sorry for the delay, after a deeper look seems a false alarm, not sure why vercel cached some files. I just noticed that `timezone` is undefined now, affecting the footer (`timezone.tsx`)

[the remark this answers is not in the document]
Comment (dennissiq) [reply pages/index.tsx:54]: @baires Oh, no problems :)

Can you approve this PR?

**Pull request merged**